models/modules/dense_motion.py

```diff
@@ -0,0 +1,54 @@
+import torch
+from torch import nn
+from torch.nn import functional as F
+
+
+class DenseMotion(nn.Module):
+    def __init__(self, in_channels, mid_channels, norm_layer, use_bias):
+        super(DenseMotion, self).__init__()
+        
+        self.grid_map = nn.Sequential(
+            nn.Conv2d(in_channels, mid_channels, kernel_size=3, stride=1, padding=1, bias=use_bias),
+            norm_layer(mid_channels),
+            nn.ReLU(True),
+            nn.Conv2d(mid_channels, 2, kernel_size=3, stride=1, padding=1, bias=use_bias),
+            nn.Tanh()
+        )
+
+    def get_grid(self, x):
+        '''
+            Function for calculating grid
+        '''
+        grid = self.grid_map(x)
+        grid = grid.permute(0, 2, 3, 1)
```

Comment:
Better to comment somewhere nearby what is the current shape like: "now it is [bs, h, w, c]"